Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/DamageTargetByRiddle.cs

```diff
@@ -0,0 +1,78 @@
+using System.Threading.Tasks;
+using System.Runtime.Versioning;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Xml.Serialization;
+using System.Xml.Schema;
+using System.Xml;
+
+public class DamageTargetByRiddle : ActionEffect
+{
+    public int minDamage { get; private set; }
+    public int maxDamage { get; private set; }
+
+    public DamageTargetByRiddle(int minDamage, int maxDamage, TargetType target)
+    {
+        this.minDamage = minDamage;
+        this.maxDamage = maxDamage;
+        Target = target;
+    }
+
+    public DamageTargetByRiddle()
+    {
+
+    }
+
+    public override void Apply(params CombatUnit[] targets)
+    {
+        foreach (CombatUnit target in targets)
+        {
+            CombatUnit attacker = CombatManager.currentCombatant;
+            int damageAmount = calculateDamageAmount(minDamage, maxDamage);
+            target.ApplyDamage(damageAmount);
+            DialogueBoxUIManager.AddStringToDialogueBox
+            ($"{DialogueBoxUIManager.FormatCombatUnitColor(attacker)}"
+            + $" dealt {DialogueBoxUIManager.FormatDamageColor(damageAmount)} damage to "
+            + $"{DialogueBoxUIManager.FormatCombatUnitColor(target)}");
+        }
+    }
+
+    public static async Task<bool> HandleRiddleResult() {
+        await Task.Delay(500);
+        System.Random randomNum = new System.Random();
+        int riddleIndex = randomNum.Next(0, SyllabusRiddleManager.Riddles.Length);
+        SyllabusRiddleUIManager.DisplayRiddle(riddleIndex);
+        while (AnswerSubmissionManager.isSubmitted == false) {
+            await Task.Delay(100);
+        }
+        AnswerSubmissionManager.isSubmitted = false;
+        return AnswerSubmissionManager.isCorrect;
+
+    }
```

Comment:
What was the inspiration for using async rather than coroutines here? If it works I'm down for it but I'm wondering what made you go with this since async methods are spooky to me lmao

Reply:
yeah, I def need to go back and use a coroutine instead. At the time, I couldn't get coroutines to work bc you cant use them with static variables/methods but async functions managed to work to add a delay. Tbh, I def should look more into the async functions bc I wouldn't be surprised if it breaks things during a full build